tests/middleware/test_middleware.py

```diff
@@ -0,0 +1,149 @@
+"""Import the Middleware class from the corresponding module"""
+from app.middleware.request_validations import Middleware
+
+
+data_dict_valid_python = {
+    "code": "print('Hello, World!')",
+    "input": "",
+    "output": "Hello, World!",
+    "time_limit": 5,
+    "memory_limit": 256,
+    "language": "python3",
+}
+
+"""Data dictionaries for the tests"""
+data_dict_valid_java = {
+    "code": "print('Hello, World!')",
+    "input": "",
+    "output": "Hello, World!",
+    "time_limit": 5,
+    "memory_limit": 256,
+    "language": "java11",
+}
+
+"""Data dictionaries for the tests"""
+data_dict_valid_cpp = {
+    "code": "print('Hello, World!')",
+    "input": "",
+    "output": "Hello, World!",
+    "time_limit": 5,
+    "memory_limit": 256,
+    "language": "cpp14",
+}
+
+"""Data dictionaries for the tests"""
+data_dict_invalid_empty_code = {
+    "code": "",
+    "input": "",
+    "output": "Hello, World!",
+    "time_limit": 5,
+    "memory_limit": 256,
+    "language": "python3",
+}
+
+"""Data dictionaries for the tests"""
+data_dict_invalid_long_time_limit = {
+    "code": "print('Hello, World!')",
+    "input": "",
+    "output": "Hello, World!",
+    "time_limit": 15,
+    "memory_limit": 256,
+    "language": "python3",
+}
+
+"""Data dictionaries for the tests"""
+data_dict_invalid_language = {
+    "code": "print('Hello, World!')",
+    "input": "",
+    "output": "Hello, World!",
+    "time_limit": 5,
+    "memory_limit": 256,
+    "language": "c",
+}
```

Comment:
Esto se podría pasar a otro archivo mejor, por ejemplo dentro de la carpeta tests un archivo que sea resources.py, y los nombres de las variables en mayúsculas, la mejor forma de hacerlo seria con un fixture de pytest https://docs.pytest.org/en/7.1.x/how-to/fixtures.html

Y creo que para no repetir es hacer un solo diccionario que pase, y en cada test modificarlo para lo que quieras que falle o probar lo especifico

Reply:
resolve in https://github.com/MinervaLMS/judge-api/commit/818f80c227114367814f43b7fdb5435c8b8618c6